### PR TITLE
Remove unused crates

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**
N / A

**Description of changes:**
After #1983, nothing used `growpart`

```
* sources: remove unused crate growpart
```
From [1709](https://github.com/bottlerocket-os/bottlerocket/pull/1709#discussion_r690906953)
> $1.00 if we delete servicedog since we haven't used it for anything in over a year

^ and that was almost 1 year ago, so we have been shipping `servicedog` for ~2 years :smile: 

```
* sources: remove unused crate servicedog
```

**Testing done:**

- Build Bottlerocket
- Booted a VM, without any failures:

```
bash-5.1# apiclient get os
{
  "os": {
    "arch": "x86_64",
    "build_id": "eccb14f5",
    "pretty_name": "Bottlerocket OS 1.8.0 (aws-dev)",
    "variant_id": "aws-dev",
    "version_id": "1.8.0"
  }
}

bash-5.1# systemctl status
● ip-172-31-61-1.us-west-2.compute.internal
    State: running
     Jobs: 0 queued
   Failed: 0 units
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/os/os.spec [hunk 1]
```diff
@@ -405,9 +392,6 @@ for p in apiclient ; do
 done
 
 install -d %{buildroot}%{_cross_sbindir}
```
Comment (bcressey) [inline packages/os/os.spec]: nit: can remove this line too

**Review by bcressey — APPROVED** [review]
Nice!

**Comment by bcressey** [comment]
Seems like I owe you $2 😀

**Comment by arnaldo2792** [comment]
( forced pushed fixed comment about the remaining line )

**Review by etungsten — APPROVED** [review]

**Pull request merged**